src/combinatorics/simple_graph.lean

```diff
@@ -206,4 +206,26 @@ by { intro v, simp }
 
 end finite
 
+section maps
+
+variables {W : Type*} (G) (H : simple_graph W)
+
+/-- A graph homomorphism maps adjacent vertices to adjacent vertices -/
+abbreviation hom := rel_hom G.adj H.adj
+
+infix ` →g ` : 50 := hom
+
+/-- A graph embedding is an embedding `f` such that for vertices `v w : V`,
+  `G.adj f(v) f(w) ↔ G.adj v w `-/
+abbreviation embedding := rel_embedding G.adj H.adj
```

Comment:
I find hook arrow notation for this to be somewhat surprising.  Having bona fide embeddings makes sense, but to me hook arrow always means monomorphism, not regular monomorphism.  I started a topic on the math stream to get some community input.

Could you add a comment that a graph embedding is a graph isomorphism onto an induced subgraph?